test/e2e/treatments.e2e.test.ts

```diff
@@ -0,0 +1,92 @@
+import 'dotenv/config.js'
+import * as mwoffliner from '../../src/mwoffliner.lib.js'
+import domino from 'domino'
+import rimraf from 'rimraf'
+import { execa } from 'execa'
+import { jest } from '@jest/globals'
+import { renderers } from './rendererList.js'
+import { zimcheckAvailable, zimdumpAvailable, zimcheck, zimdump } from '../util.js'
+
+jest.setTimeout(200000)
+
+let zimcheckIsAvailable
+let zimdumpIsAvailable
+
+beforeAll(async () => {
+  zimcheckIsAvailable = await zimcheckAvailable()
+  zimdumpIsAvailable = await zimdumpAvailable()
+})
+
+const setDefaultParams = (renderName: string, testId: string) => {
+  return {
+    mwUrl: 'https://en.wikipedia.org',
+    adminEmail: 'test@kiwix.org',
+    outputDirectory: testId,
+    redis: process.env.REDIS,
+    articleList: 'User:Kelson/MWoffliner_CI_reference',
+    forceRender: renderName,
+  }
+}
```

Comment:
Same here, we should have a function just doing straight the job and returning the `outFiles`. The variables would simply be the `articleList` and the `renderName`, maybe the `mwUrl` as optional parameter. This file should really be reduced to the bar minimum of line with only the interesting stuff which is basically the `commonTreatmentTest()`.

At the end we might just achieve to have one call with the test in callback and that's all?!

Reply:
I've created `getOutFiles()` function where I keep all the logic about creating ZIM and returning `outFiles`, pls check.

Reply:
OK, but at the same time you see, already in that PR, that for many e2e tests you need to access to the `dump` object! So this is is only soving the problem halway.

Why not having something like `testWithAllRenders(mwUrl, format, ..., callback{})` and in the callback you get access to `outFiles` to `dump` and whatever is necessary for e2e test to be done specifically.

e2e should only contain their very specific lines and nothing else.

This wish is actually for everything, please factorise your code the most optimal manner. The less code it is the better.

Reply:
I faced the problem that I couldn't nest a test block inside another test or callback function in Jest. For comparison purpose I created **test/e2e/en.e2e-callback.test.ts** with **test/testAllRendersTemplateCallback.ts** and  **test/e2e/en.e2e.test.ts** with **test/testAllRendersTemplate.ts**. Pros and cons:
1. test/testAllRendersTemplate.ts - have to keep all the treatment tests inside but related **test/e2e/en.e2e.test.ts** is pretty small.
2. **test/testAllRendersTemplateCallback.test.ts** - returns only the outFile and overall has a single responsibility. Cons: you can't avoid forEach loop in related **test/e2e/en.e2e-callback.test.ts** and other e2e tests that will use this template.

Reply:
Callback inside for..of renders loop has been implemented **test/testAllRenders.ts**